File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-wishlist/src/services/wishlistService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { WishlistPageResult } from "../types/wishlistModels.js";

/**
 * Wishlist service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface WishlistService {
  readonly capability: "wishlist";
}

export function createWishlistService(client: AppstoreAppClient): WishlistService {
  void client;
  return { capability: "wishlist" };
}

export function emptyWishlistResult(): WishlistPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_wishlist/lib/src/services/wishlist_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/wishlist_models.dart';

/// Wishlist service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class WishlistService {
  const WishlistService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'wishlist';

  WishlistPageResult<Never> empty() =>
      const WishlistPageResult<Never>(items: <Never>[]);
}
